rpc_sidecar/src/lib.rs

```diff
@@ -116,6 +118,14 @@ fn resolve_address(address: &str) -> anyhow::Result<SocketAddr> {
         .ok_or_else(|| anyhow::anyhow!("failed to resolve address"))
 }
 
+fn encode_request(req: &BinaryRequest, id: u16) -> Result<Vec<u8>, bytesrepr::Error> {
+    let header = BinaryRequestHeader::new(SUPPORTED_PROTOCOL_VERSION, req.tag(), id);
```

Comment:
So this is a header that has both SUPPORTED_PROTOCOL_VERSION and the request id? Shouldn't it be a wrapper over a vector of headers? A tuple (u8, bytes[]) where the u8 would be a header tag? That would give us flexibility of adding more headers in the future. If we don't do it like that - we're painting ourselves into a corner, aren't we?